src/ParseServer.js

```diff
@@ -170,11 +172,28 @@ class ParseServer {
     });
     const filesController = new FilesController(filesControllerAdapter, appId);
 
+    const pushQueueOptions = (push || {}).queueOptions || {};
+
+    if ((push || {}).queueOptions) {
+      delete push.queueOptions;
+    }
     // Pass the push options too as it works with the default
-    const pushControllerAdapter = loadAdapter(push && push.adapter, ParsePushAdapter, push || {});
-    // We pass the options and the base class for the adapter,
+    const pushAdapter = loadAdapter(push && push.adapter, ParsePushAdapter, push || {});
+    // We pass the options and the base class for the adatper,
     // Note that passing an instance would work too
-    const pushController = new PushController(pushControllerAdapter, appId, push);
+    const pushController = new PushController();
+
+    const hasPushSupport = pushAdapter && push;
+
+    const {
+      disablePushWorker
+    } = pushQueueOptions;
```

Comment:
ugh, can you help me out with this construct?  what the difference from simply `const disablePushWorker = pushQueueOptions;`